### PR TITLE
bring ByteBufferOutputStream from fmj

### PR DESCRIPTION
**Pull request by bbaldino**
this is part of the effort to remove the fmj dependency

### CODE REVIEW

File: src/org/jitsi/util/ByteBufferOutputStream.java [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+     * {@inheritDoc}
+     */
+    @Override
+    public void write(int b)
+        throws IOException
+    {
+        if (index >= endIndex)
+        {
+            throw new IOException(
+                "This " + net.sf.fmj.utility.ByteBufferOutputStream.class.getName()
```
Comment (ibauersachs) [inline src/org/jitsi/util/ByteBufferOutputStream.java]: This isn't an FMJ class anymore.

**Comment by ibauersachs** [comment]
> this is part of the effort to remove the fmj dependency

Maybe I missed something because I'm not always in The Call. Could you please clarify that? AFAIK FMJ is the core port for rendering all audio/video (in desktop applications).

File: src/org/jitsi/util/ByteBufferOutputStream.java (hunk 1 above)
Comment (bbaldino) [reply src/org/jitsi/util/ByteBufferOutputStream.java]: ah good catch, will get this fixed

**Comment by bbaldino** [comment]
> > this is part of the effort to remove the fmj dependency
> 
> Maybe I missed something because I'm not always in The Call. Could you please clarify that? AFAIK FMJ is the core port for rendering all audio/video (in desktop applications).

We no longer need that functionality in the bridge and it's making some other work difficult, so the idea is to implement the rtp stack logic that's needed by the bridge in libjitsi so we can remove the dependency on fmj there.  from talking with emil, i think we should then be able to change fmj to rely on libjitsi for the rtp stack so we don't have any duplication.  other components which need fmj can keep using it for the encoding/decode logic.

**Comment by ibauersachs** [comment]
Fine, it might even improve some broken parts. But please make sure that Jitsi Desktop still works. I don't expect you guys to actively support it, but please don't break it either. I can't afford maintaining a fork of libjitsi.

**Comment by bgrozev** [comment]
Jigasi and jitsi-hammer depend on fmj and there are no plans to remove that dependency, so I don't think there's any risk for the desktop client.

**Pull request merged**